sqlglot/expressions.py

```diff
@@ -3673,25 +3673,57 @@ def table_(table, db=None, catalog=None, quoted=None, alias=None) -> Table:
     )
 
 
-def values(values, alias=None) -> Values:
+def values(
+    values: t.Sequence[t.Tuple[str | Expression, ...]],
+    alias: t.Optional[str] = None,
+    columns_and_types: t.Optional[t.List[str | t.Tuple[str, str | DataType]]] = None,
+) -> Values:
     """Build VALUES statement.
 
     Example:
         >>> values([(1, '2')]).sql()
         "VALUES (1, '2')"
 
     Args:
-        values (list[tuple[str | Expression]]): values statements that will be converted to SQL
-        alias (str): optional alias
-        dialect (str): the dialect used to parse the input expression.
-        **opts: other options to use to parse the input expressions.
+        values: values statements that will be converted to SQL
+        alias: optional alias
+        columns_and_types: Optional list of ordered column names or column names and types. If types are provided,
+            they will be used to cast the values. An alias is required when providing column names.
 
     Returns:
         Values: the Values expression object
     """
+    if columns_and_types and not alias:
+        raise ValueError("Alias is required when providing columns_and_types")
+    column_names = []
+    column_types = []
+    if columns_and_types is not None:
+        for column in columns_and_types:
+            if isinstance(column, tuple):
+                column_names.append(to_identifier(column[0]))
+                if isinstance(column[1], str):
+                    column_types.append(DataType.build(column[1]))
+                else:
+                    column_types.append(column[1])
+            else:
+                column_names.append(to_identifier(column))
+    if column_types:
+        casted_rows = []
+        for row in values:
```

Comment:
hmm… you’re just doing a cast on everything? im not sure we should do that, it’s more efficient to just nest this in a select and cast it, so we may not need this api?

Reply:
no because the sql string is way bigger right? let’s chat tomorrow about this

Reply:
Right the size of the expression is a fair concern. I think we should still make it an option for users although we might in other contexts decide to not use it and cast the select.